src/main/java/net/htmlcsjs/htmlTech/api/laserpipe/LaserPipeRenderer.java

```diff
@@ -0,0 +1,268 @@
+    }
+
+    public void registerIcons(TextureMap map) {
+        for (LaserPipeType laserPipeType : LaserPipeType.values()) {
+            ResourceLocation inLocation = new ResourceLocation(htmlTech.MODID, String.format("blocks/pipe/laser_%s", laserPipeType.name));
+            ResourceLocation sideLocation = new ResourceLocation(htmlTech.MODID, String.format("blocks/pipe/laser_%s", laserPipeType.name));
+
+            TextureAtlasSprite inTexture = map.registerSprite(inLocation);
+            TextureAtlasSprite sideTexture = map.registerSprite(sideLocation);
+            this.pipeTextures.put(laserPipeType, new PipeTextureInfo(inTexture, sideTexture));
+        }
+    }
+
+    @SubscribeEvent
+    public void onModelsBake(ModelBakeEvent event) {
+        GTLog.logger.info("registering laser pipe model");
+        event.getModelRegistry().putObject(MODEL_LOCATION, this);
+    }
+
+    @Override
+    public void renderItem(ItemStack rawItemStack, ItemCameraTransforms.TransformType transformType) {
+        ItemStack stack = ModCompatibility.getRealItemStack(rawItemStack);
+        if (stack.getItem() instanceof ItemBlockLaserPipe) {
+            CCRenderState renderState = CCRenderState.instance();
+            GlStateManager.enableBlend();
+            renderState.reset();
+            renderState.startDrawing(7, DefaultVertexFormats.ITEM);
+            BlockLaserPipe blockFluidPipe = (BlockLaserPipe) ((ItemBlockLaserPipe) stack.getItem()).getBlock();
+            LaserPipeType pipeType = blockFluidPipe.getItemPipeType(stack);
+            if (pipeType != null) {
+                int connections = 1 << EnumFacing.SOUTH.getIndex() | 1 << EnumFacing.NORTH.getIndex() | 1 << 6 + EnumFacing.SOUTH.getIndex() | 1 << 6 + EnumFacing.NORTH.getIndex();
+                connections |= 4096;
+                this.renderPipeBlock(pipeType, 0xFFFFFF, renderState, new IVertexOperation[0], connections);
+            }
+
+            renderState.draw();
+            GlStateManager.disableBlend();
+        }
+    }
+
+    public boolean renderBlock(IBlockAccess world, BlockPos pos, IBlockState state, BufferBuilder buffer) {
+        CCRenderState renderState = CCRenderState.instance();
+        renderState.reset();
+        renderState.bind(buffer);
+        renderState.setBrightness(world, pos);
+        IVertexOperation[] pipeline = {new Translation(pos)};
+
+        BlockLaserPipe blockFluidPipe = ((BlockLaserPipe) state.getBlock());
+        TileEntityLaserPipe tileEntityPipe = (TileEntityLaserPipe) blockFluidPipe.getPipeTileEntity(world, pos);
+
+        if (tileEntityPipe == null) {
+            GTLog.logger.info("Tile is null");
+            return false;
+        }
+
+        LaserPipeType fluidPipeType = tileEntityPipe.getPipeType();
+        int paintingColor = tileEntityPipe.getInsulationColor();
+        int connectedSidesMap = blockFluidPipe.getVisualConnections(tileEntityPipe);
+
+        if (fluidPipeType != null) {
```

Comment:
This must be the reason the block doesn't render, but I don't know why it is null